saviSimulation/savi/jason_processing/Geometry.java

```diff
@@ -82,4 +78,25 @@ public static PVector absolutePositionFromPolar(double azimuth, double elevation
 		PVector absolutePosition = new PVector(x,y,z);
 		return absolutePosition;
 	}
-}
+
+	public static double normalize02PI(double angle) {
+		while (angle < 0)                                                                                    
+			angle += (2 * Math.PI);
+		
+		while (angle > (2 * Math.PI))
+			angle -= (2 * Math.PI);
+	
+		return angle;
+	}
+
+	public static double normalizeMinusPIPI(double angle) {
+		while (angle < -Math.PI)                                                                                    
+			angle += (Math.PI);
+		
```

Comment:
shouldn't this be += 2* Pi ?
Otherwise it doesn't produce an equivalent angle.

Also, an object at azimuth 0 and elevation 3/4 PI is also at azimuth PI and elevation 1/4 PI.  Given this position in cartesian coordinates, which of the polar coordinates will our conversion produce?

Reply:
So I investigated and I can answer my own question:

from Geometry Unit tests:
assertTrue(((-Math.PI/2) <= elevation) && (elevation <= (Math.PI/2)));	// elevation must be between -PI/2 and PI/2

Reply:
But normalizing elevation and azimuth angles means you need to do two changes:
Assume input (az, el) is (0, 3/4 PI), output should be (pi, pi/4).

Reply:
Elevation is supposed to represent the angle that the object is above or below your XY plane, regardless of what direction you are facing. This means that it can never be less than -PI (directly below) or greater then PI (right above you). Otherwise, this is represented be a difference in the azimuth angle.

Reply:
@gtrabes @gtrabes please see if my fixes make sense

Reply:
@pgavigan 
I think directly above is +PI/2 and directly below is -PI/2

Reply:
Should line 111 of Geometry.java be `elevation2 = -Math.PI + elevation2;`? (note the change in + vs -.

Reply:
Also, @ntartania, you are correct, PI/2 is directly above.

Reply:
> Should line 111 of Geometry.java be `elevation2 = -Math.PI + elevation2;`? (note the change in + vs -.

No: if elevation2 = -3/4 PI, then -PI - elevation2 = -PI - (-3/4 PI) = -PI + 3/4 = -PI/4, which is what we aimed to obtain.

Reply:
> > Should line 111 of Geometry.java be `elevation2 = -Math.PI + elevation2;`? (note the change in + vs -.
> 
> No: if elevation2 = -3/4 PI, then -PI - elevation2 = -PI - (-3/4 PI) = -PI + 3/4 = -PI/4, which is what we aimed to obtain.

Right, I was forgetting about the second negative.